reactioncommerce/reaction-next-starterkit
# Authentication with OIDC (OAuth2) to reaction hydra

Resolves #272
Impact: **major**
Type: **feature**

## Issue

Wire-up authentication

## Solution

Use passport.js with the oauth2 strategy

## Breaking changes

none

## Testing

1. Clear your schedule
1. clone `https://github.com/reactioncommerce/reaction-platform`
1. `cd reaction-platform`
1. `make`
1. wait a while, it will download all repos needed
1. `cd reaction && git checkout release-2.0.0`
1. `cd reaction-next-starterkit && git checkout feat-272-mikemurray-oidc`
1. `make start`
1. wait a while
1. Open the starter kit `http://localhost:4000`
1. Open the login dropdown and "Sign In"
1. You should be redirected to a Reaction login page.
1. Sign into an account
1. You should be redirected back to the starter kit
1. Open the account dropdown and you should see the name of the signed in user
